schemas/com.snowplowanalytics.badrows/storage_loader_failure/jsonschema/1-0-0

```diff
@@ -0,0 +1,73 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a storage loader failure bad row",
+  "self": {
+    "vendor": "com.snowplowanalytics.badrows",
+    "name": "storage_loader_failure",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "failure": {
+      "type": "object",
+      "description": "Information regarding the storage loader failure",
+      "properties": {
+        "schema": {
+          "type": "string",
+          "description": "Schema of the particular storage loader failure",
+          "pattern": "^iglu:[a-zA-Z0-9-_.]+/[a-zA-Z0-9-_]+/[a-zA-Z0-9-_]+/[0-9]+-[0-9]+-[0-9]+$"
+        },
+        "data": {
+          "type": "object",
+          "description": "Data associated with the particular storage loader failure"
+        }
+      },
+      "required": [ "schema", "data" ],
+      "additionalProperties": false
+    },
+    "payload": {
+      "type": "object",
+      "description": "The collector payload that resulted in a storage loader failure",
+      "properties": {
+        "enrichedEvent": {
+          "type": "string",
+          "description": "Enriched event associated with this failure"
+        },
```

Comment:
should we adopt a more extensive data model like https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/elasticsearch_enriched_event/jsonschema/2-0-0

Reply:
Yes - think so! Was your initial proposal that this be the enriched event tsv? 
I'd love to brainstorm what the format of this should be, and how that relates to the format of the data outputted at enrichment failure. (Would be nice if it was a comparable format, but lacking the fields that would be computed at enrichment.)

Reply:
Also - we want these failures to be very easy to recover i.e. to be able to write a simple process where our different loaders can quickly retry loading the bad data. Not sure how this requirement impacts the design decisions here about how the data is stored... (Maybe we attach two copies to the bad row - one that reflects the format of the data prior to loading to the specific storage target format, and another that is the "canonical" format?)

Reply:
Unfortunately, there's an important corner case that we must be aware of. Despite the fact that data in 99% of cases was already enriched and we *can* assume the enriched format there's also a chance that something sends rubbish to PubSub/Kafka topic or S3 bucket and Loader simply cannot read it. I'm not sure whether this should be another member of union type

```
StorageLoaderFailure = AbsolutelyInvalidData | ProblemWithEnrichedData
```

Or maybe even totally separate type of bad row (quite undesirable), but we cannot **always** assume an enriched format and pretty much nothing apart from custom loader's message.

Reply:
Since this one is completely unopinionated, we could have `Payload = EnrichedEvent | SomeOtherThing`